system/jhtree/keybuild.cpp

```diff
@@ -201,8 +204,13 @@ class CKeyBuilderBase : public CInterface
         if (out)
         {
             out->flush();
-            out->seek(0, IFSbegin);
-            keyHdr->write(out, crc);
+            if (keyHdr->getKeyType() & USE_TRAILING_HEADER)
+                keyHdr->write(out, crc);  // write a copy at end too, for use on systems that can't seek
+            if (!(keyHdr->getKeyType() & TRAILING_HEADER_ONLY))
+            {
+                out->seek(0, IFSbegin);
+                keyHdr->write(out, crc);
```

Comment:
If the trailing header isn't written then the crc from the original header write needs to be used.

Reply:
Worse than that - also need to include the crc of the trailing header.

Reply:
the key header contains the crc for the whole file, it's just returning the crc of the KeyHdr node here, to roll it up with the crc's of the node data, to return the complete file crc for the file to the caller to publish it in Dali.

I think if you calc. once (pass null into one of the write calls if both are called), you can then merge in this crc before and after the node data crc in finish().

Reply:
@jakesmith It's worse than that. I think I would have to return crc info for the original head info if I don't overwrite it, or new head if I do (to merge it in properly), and also ensure crc info for trailing header (if I write one) is folded in